dvc/repo/diff.py

```diff
@@ -1,246 +1,56 @@
-from dvc.scm.git import DIFF_EQUAL
 from . import locked
+from ..compat import fspath
 
 
-DIFF_TARGET = "target"
-DIFF_IS_DIR = "is_dir"
-DIFF_OLD_FILE = "old_file"
-DIFF_OLD_CHECKSUM = "old_checksum"
-DIFF_NEW_FILE = "new_file"
-DIFF_NEW_CHECKSUM = "new_checksum"
-DIFF_SIZE = "size_diff"
-DIFF_DEL = "del"
-DIFF_IDENT = "ident"
-DIFF_CHANGE = "changes"
-DIFF_NEW = "new"
-DIFF_MOVE = "moves"
-DIFF_LIST = "diffs"
-DIFF_SIZE_UNKNOWN = "?"
-DIFF_A_OUTPUT = "a_output"
-DIFF_B_OUTPUT = "b_output"
-DIFF_DELETED = "deleted_file"
-DIFF_IS_NEW = "created_file"
+Diffable = collections.namedtuple("Diffable", "filename, checksum, size")
+Diffable.__doc__ = "Common interface for comparable entries."
 
 
-def _file_not_exists(error, result):
-    if error.errno == ENOENT:
-        result.update({DIFF_SIZE: DIFF_SIZE_UNKNOWN})
-    else:
-        raise error
-
-
-def _extract_dir(self, dir_not_exists, output):
-    """Extract the content of dvc tree file
-    Args:
-        self(object) - Repo class instance
-        dir_not_exists(bool) - flag for directory existence
-        output(object) - OutputLOCAL class instance
-    Returns:
-        dict - dictionary with keys - paths to file in .dvc/cache
-                               values -checksums for that files
-    """
-    if not dir_not_exists:
-        lst = output.dir_cache
-        return {i["relpath"]: i["md5"] for i in lst}
-    return {}
-
-
-def _get_dir_info(dir_not_exists, a_output):
-    if not dir_not_exists:
-        return str(a_output), a_output.checksum
-    return "", ""
-
-
-def _ident_files(a_entries, b_entries):
-    keys = [
-        key for key in a_entries.keys() if b_entries.get(key) == a_entries[key]
-    ]
-    return len(keys)
-
-
-def _modified_files(self, a_entries, b_entries):
-    keys = [key for key in a_entries.keys() if key in b_entries]
-    diff_size = 0
-    modified_count = 0
-    for key in keys:
-        if a_entries[key] != b_entries[key]:
-            modified_count += 1
-            diff_size += os.path.getsize(
-                self.cache.local.get(b_entries[key])
-            ) - os.path.getsize(self.cache.local.get(a_entries[key]))
-    return modified_count, diff_size
-
-
-def _deleted_files(self, a_entries, b_entries):
-    diff_size = 0
-    deleted_count = 0
-    for key, value in a_entries.items():
-        if key not in b_entries:
-            deleted_count += 1
-            diff_size -= os.path.getsize(self.cache.local.get(a_entries[key]))
-    return deleted_count, diff_size
-
-
-def _new_files(self, a_entries, b_entries):
-    diff_size = 0
-    new_count = 0
-    for key, value in b_entries.items():
-        if key not in a_entries:
-            new_count += 1
-            diff_size += os.path.getsize(self.cache.local.get(b_entries[key]))
-    return new_count, diff_size
-
+def diffable_from_output(output):
+    try:
+        size = os.path.getsize(fspath(output.path_info))
+    except OSError:
```

Comment:
You need to ignore ENOENT (or the corersponding exception, not OSERrror in general)